### PR TITLE
Define function to get specific vertex of polytope

### PR DESCRIPTION
**Pull request by mfsch**
As discussed in #406, this defines a new function to get a specific vertex of a polytope. It also implements the `vertices` function of the Box in terms of this new function, generalizing it to N dimensions.

A few open questions:
- Maybe it is better to leave the implementation of `vertices(polytope)` as it was? The new implementation would allow new polytopes to automatically get the implementation of `vertices` if they define `vertex` and `nvertices` and should not have any runtime overhead since the conditional can be resolved at compile-time, but has circular dependencies if those two methods are not defined.
- For the Box, I think it would be cleaner to define the vertex order to consistently switch between the lower and upper boundary along the first dimension, i.e. remove the conditional at lines 61–63 (e.g. [(0,0),(1,0),(0,1),(1,1)] in 2D). This would require a small change in the `boundary` function. However, I’m not sure if there is any other code that relies on the order of the box vertices, so maybe it’s better to keep the order as it is?
- Is there a reason that the Box is classified as a primitive instead of a polytope?

### CODE REVIEW

File: src/polytopes.jl [hunk 1]
```diff
@@ -38,12 +38,25 @@ Return the parametric dimension or rank of the polytope.
...
 Return the vertices of a `polytope`.
 """
-vertices(p::Polytope) = p.vertices
+function vertices(p::Polytope)
+  if hasfield(typeof(p), :vertices)
+    p.vertices
+  else
+    collect(vertex(p, ind) for ind in 1:nvertices(p))
+  end
+end
```
Comment (juliohm) [inline src/polytopes.jl:59]:
```suggestion
vertices(p::Polytope) = [vertex(p, ind) for ind in 1:nvertices(p)]
```

**Comment by juliohm** [comment]
> * Is there a reason that the Box is classified as a primitive instead of a polytope?

I love it how you are spotting inconsistencies in our codebase ❤️ I realized that all our `Primitive` types do not define the `vertices` function, except for `Box`, which is an indication that maybe it should be a `Polytope` instead. Originally, the `Box` was considered a `Primitive` because of its "lazy" representation where we could save memory and build the vertices on the fly, but I am starting to think that it could simply be a `Polytope` for greater consistency.

We already have "boxes" that are more flexible like `Segment`, `Quadrangle` and `Hexahedron` for 1D, 2D, and 3D, respectively. They are more flexible in the sense that they do not need to be axis-aligned nor they need to have right angles.

Can we go one step back and investigate the possibility of `Box` as a `Polytope`? If you could draft a PR where this change is made, we could assess the impact in algorithms downstream. We could then come back to the issue of the new `vertex` function proposed in this present PR.

In summary:

1. Move `Box <: Primitive` to `Box <: Polytope` and see if all tests pass.
2. Define the new `vertex` function for `Polytope` as its done in this PR.
3. Investigate the order of vertices in `Box` in a third PR to see if tests pass.

Sorry for requesting so many different PRs, it is because you're touching on very fundamental design issues that we didn't realize, and so I think it is important to handle them separately before continuing with the changes in these opened PRs 🙏🏽

**Comment by juliohm** [comment]
I am performing some investigation with the Box type to see if it is worth simply eliminating the `vertices` implementation for it. Will let you know when the investigation is concluded.

**Comment by juliohm** [comment]
@mfsch I removed the `vertices` implementation of `Box`, it is not internal to the `boundary` definitions. Can you please rebase this PR on top of the latest master branch?

**Comment by juliohm** [comment]
@mfsch I will close this PR given that it is changing Box and we refactored it in the master branch. Let's see if there is still the need for a `vertex` function or if it was just a design issue with Box that motivated its creation.
